docs/ADR/2019-11-27-firestore-saving-timed-transcripts.md

```diff
@@ -0,0 +1,406 @@
+
+article from [coderwall, compress your json data](https://coderwall.com/p/mekopw/jsonc-compress-your-json-data-up-to-80)
+
+```js
+const fs = require('fs')
+const jsonc = require('jsonc');
+const gcpSttResponseJson = require('./create-1574786669322.json')
+
+const gcpToDpe = require('gcp-to-dpe');
+const gcpTranscript = gcpSttResponseJson.response;
+const dpeTranscript = gcpToDpe(gcpTranscript);
+fs.writeFileSync('dpe.json',JSON.stringify(dpeTranscript,null,2))
+
+
+const compressedJSON = jsonc.parse( dpeTranscript );
+fs.writeFileSync('jsonc-compressed.json',compressedJSON);
+```
+got an error
+```
+return result + (insideComment ? strip(jsonString.slice(offset)) : jsonString.slice(offset));
+
+TypeError: jsonString.slice is not a function
+```
+
+<!-- * Good, because [argument a]
+* Good, because [argument b] -->
+* Bad, because wasn't able to try out the libary
+
+### 7. Compression - binary with UBJSON
+[ubjson](https://www.npmjs.com/package/@shelacek/ubjson)
+
+Firestore can support binary data, so json could be compressed to binary.
+
+When trying this out 
+```js
+const fs = require('fs')
+const ubjson = require('@shelacek/ubjson');
+const gcpToDpe = require('gcp-to-dpe');
+
+const gcpSttResponseJson = require('./create-1574786669322.json')
+const gcpTranscript = gcpSttResponseJson.response;
+const dpeTranscript = gcpToDpe(gcpTranscript);
+
+const buffer = ubjson.encode({ dpeTranscript, from: ['UBJSON'] });
+
+fs.writeFileSync('Ubjson',buffer);
+```
+
+Got an error
+```
+RangeError: Maximum call stack size exceeded
+```
+<!-- * Good, because [argument a]
+* Good, because [argument b] -->
+* Bad, because wasn't able to try out the libary
+
+
+
+### 8. Convert to a collection of 3 `tsv` documents 
+
+It is stated in firebase docs, that within 1mb, you could hold a whole book (soruce needed). As generally a whole book in plain text it's under 1mb.
+[Moby dick from project gutenberg]((https://www.gutenberg.org/files/2701/2701-h/2701-h.htm)) it's 1.3mb. roughly 215,831 words.
+
+If 3581 words in 20 minutes then in 215,831 words it's 1,205.42 minutes, which is roughly 20 hours.
+
+_if my math is not failing me_
+
+So words like this 
+
+```json
+{
+  "words": [
+    {
+      "id": 0,
+      "start": 1.4,
+      "end": 3.9,
+      "text": "Can"
+    },
+    {
+      "id": 1,
+      "start": 3.9,
+      "end": 4,
+      "text": "you"
+    },
+    {
+      "id": 2,
+      "start": 4,
+      "end": 4.1,
+      "text": "hear"
+    },
+    {
+      "id": 3,
+      "start": 4.1,
+      "end": 4.2,
+      "text": "it?"
+    },
+    ...
+```
+
+could be converted to one string, so a document in a collection
+```
+Text startTime endTime\ttext startTime endTime\t...
+```
+
+```
+Can 1.4 3.9\tyou 3.9 4\thear 4 4.1\tit? 4.1 4.2\t...
```

Comment:
This isn't really a tsv:
```
Text startTime endTime\ttext startTime endTime\t...
```
If it's a tsv, it would look like this:
```csv
Text\tstartTime\tendTime
Text\tstartTime\tendTime
...
```
Which would be represented as a string like so programattically:
`Can\1.4\t3.9\nyou\t3.9\t4\n`

You generally use the \n to break to say that it's a new record, but you use tabs to differentiate the values, instead of spaces.
See [JS TSV lib](https://github.com/ricardobeat/TSV)

Reply:
ah, yeah good catch